doc/ui-guidelines.md

```diff
@@ -0,0 +1,257 @@
+> DO NOT USE anon functions directly in the props
+ 
+BAD
+```clojure
+(defn view
+  []
+  (let [[pressed? set-pressed] (rn/use-state false)]
+    [rn/pressable
+     {:style               (style/main pressed?)
+      :on-press-in         #(set-pressed true)
+      :on-press-out        #(set-pressed nil)}]))
+```
+
+## Global state and subscriptions
+
+For global state management, we utilize Re-frame subscriptions. They can be likened to React state. To obtain the state, `(rf/sub [])` is employed, and to modify it, `(rf/dispatch [])` is utilized. However, they update components in a similar manner to React states.
+
+```clojure
+(defn view
+  [{:keys [selected-tab]}]
+  (let [collectible-list (rf/sub [:wallet/all-collectibles])
+        on-collectible-press (rn/use-callback
+                              (fn [{:keys [id]}]
+                               (rf/dispatch [:wallet/get-collectible-details id])))]
+    [rn/view {:style style/container}
+     (case selected-tab
+       :assets       [assets/view]
+       :collectibles [collectibles/view {:collectibles         collectible-list
+                                         :on-collectible-press on-collectible-press}])
+       [activity/view]]))
+```
+
+## Regular atoms
+
+In some cases regular atoms were used in the components
+
+
+BEFORE:
+```clojure
+(defn view
+  []
+  (let [focused? (atom false)]
+    (fn []
+      (let [on-clear               (fn []
+                                     (reset! status (if @focused? :active :default)))
+            on-focus               (fn []
+                                     (reset! focused? true))
+            on-blur                (fn []
+                                     (reset! focused? false))]))))
+```
+
+NOW:
+```clojure
+(defn view
+  []
+  (let [[focused? set-focused] (rn/use-state false)
+        on-clear               (rn/use-callback
+                                (fn []
+                                  (set-status (if focused? :active :default)))
+                                [focused?])
+        on-focus               (rn/use-callback
+                                (fn []
+                                  (set-focused true)))
+        on-blur                (rn/use-callback
+                                (fn []
+                                  (set-focused false)))]))
+```
+
+> [!IMPORTANT]
+> Instead of a regular atom, we also use a state hook. It's important to understand that its value is now encapsulated within the function, and we must pass it as a dependency to `use-callback` so that the function re-executes with the new value.
+
+## Effects
+
+In most cases, the effect should be used for implementing visual reactions (such as animations) to state changes or for lifecycle methods.
+
+ANIMATION:
+```clojure
+(defn f-zoom-button
+  [{:keys [selected? current-zoom]}]
+  (let [size (reanimated/use-shared-value (if selected? 37 25))]
+    (rn/use-effect #(reanimated/animate size (if selected? 37 25)) [current-zoom])
```

Comment:
Hi @flexsurfer, shouldn't we also update the animation when we are changing the corresponding state, instead of using the state as a dependency and then updating the animation

Reply:
And if the purpose of state is just animation, then better to just avoid it

Reply:
thank you @Parveshdhull very good point, let's discuss this, because i don't have much experience in that part, i know you working lately on improving this, so feel free to add your version

Reply:
@Parveshdhull i updated the do, pls check